src/SoftRobots/component/camera/PointCloudProcessing.cpp

```diff
@@ -0,0 +1,537 @@
+//*******************************************************************************
+//*                              SOFA :: Framework                              *
+//*                                                                             *
+//* Authors: The SOFA Team (see Authors.txt)                                    *
+//*                                                                             *
+//* Contact information: contact@sofa-framework.org                             *
+//******************************************************************************/
+#include "PointCloudProcessing.h"
+
+
+#include <sofa/core/ObjectFactory.h>
+#include <sofa/simulation/AnimateEndEvent.h>
+#include <sofa/helper/accessor.h>
+
+#include <pcl/point_types.h>
+#include <pcl/visualization/pcl_visualizer.h>
+#include <pcl/filters/passthrough.h>
+#include <pcl/filters/voxel_grid.h>
+#include <pcl/filters/radius_outlier_removal.h>
+
+#include <pcl/filters/conditional_removal.h>
+#include <pcl/segmentation/region_growing_rgb.h>
+#include <pcl/segmentation/grabcut_segmentation.h>
+
+#include <pcl/segmentation/sac_segmentation.h>
+#include <pcl/segmentation/extract_clusters.h>
+
+#include <pcl/kdtree/kdtree_flann.h>
+#include <pcl/search/search.h>
+#include <pcl/search/kdtree.h>
+#include <pcl/common/common.h>
+#include <pcl/common/angles.h>
+#include <pcl/common/transforms.h>
+#include <pcl/features/normal_3d.h>
+
+#include <pcl/point_types.h>
+#include <pcl/surface/mls.h>
+
+#include <vector>
+#include <math.h>
+#include <iostream>
+#include <iomanip>
+#include <fstream>
+#include <Eigen/Dense>
+
+using sofa::helper::ReadAccessor;
+using sofa::helper::WriteOnlyAccessor;
+
+namespace sofa
+{
+
+namespace component
+{
+
+namespace pointcloudprocessing
+{
+
+using namespace Eigen;
+using namespace std;
+
+boost::shared_ptr<pcl::visualization::PCLVisualizer> createPointCloudViewer (pcl::PointCloud<pcl::PointXYZRGB>::ConstPtr cloud_filtered);
+
+PointCloudProcessing::PointCloudProcessing() :
+    d_effectorPositions(initData(&d_effectorPositions, "effectorPositions", "Position of effectors.")),
+    d_goalPositions(initData(&d_goalPositions, "goalPositions", "Position of the contact forces.")),
+    d_normalDirections(initData(&d_normalDirections, "normalDirection", "The normal directions of effectors")),
+    d_contactLocations(initData(&d_contactLocations, "contactLocations", "The index of contact locations"))
+{
+    pclcamera.initCamera();
+}
+
+
+void PointCloudProcessing::update()
+{
+    pcl::PointCloud<pcl::PointXYZRGB>::Ptr filteredpoints = pclcamera.updatePcl();
+
+    //user-defined
+    //The following vector and matrix are the calibration matrix between the camera coordinate system and the robot coordinate system.
+    double x0 = -14.18895169; //unit is millimeter
+    double y0 = 35.32077316; //unit is millimeter
+    double z0 = 578.52358217; //unit is millimeter
```

Comment:
I guess it's only supposed to work with a specific configuration...
Can we fix that before merging...? Add a data maybe?